### ISSUE TITLE
fping6: -H option does not work: "setting time to live: Protocol not available"

### ISSUE THREAD

**Issue by little-dude** [schweikert/fping#111]
Hi,

First, thanks a lot for your work on `fping` :)

It seems that the `-H` option, used to set the TTL is broken for IPv6. I have the same issue in two different environments:

- Alpine Linux (custom `4.4.45` kernel), with `fping6` `v3.15`
- Archlinux (`4.9.6` kernel), with `fping6` `v3.13`

Here is an example on Alpine:
```
# fping6 -Ieth0 -c5 -H1 2001:3:1::4813:3585
setting time to live: Protocol not available
```
Packets are sent with a hop limit of 64:
```
16:31:14.556887 00:00:48:13:35:80 > 68:54:ed:00:3b:ee, ethertype IPv6 (0x86dd), length 118: (flowlabel 0x46e12, hlim 64, next-header ICMPv6 (58) payload length: 64) 2001:3:6::4813:3580 > 2001:3:1::4813:3585: [icmp6 sum ok] ICMP6, echo request, seq 0
16:31:14.611878 68:54:ed:00:3b:ee > 00:00:48:13:35:80, ethertype IPv6 (0x86dd), length 118: (class 0x88, flowlabel 0xfd66e, hlim 63, next-header ICMPv6 (58) payload length: 64) 2001:3:1::4813:3585 > 2001:3:6::4813:3580: [icmp6 sum ok] ICMP6, echo reply, seq 0
```

**Comment by little-dude** [comment]
I don't really understand, because this seems totally fine to me: https://github.com/schweikert/fping/blob/84ba2dd04993ac01b96c20b380562b82f66fc8a8/src/fping.c#L604

```c
if (socket6) {
            if (setsockopt(socket6, IPPROTO_IPV6, IPV6_UNICAST_HOPS, &ttl, sizeof(ttl))) {
                perror("setting time to live");
            }
}
```

According to the man:

>        IPV6_UNICAST_HOPS
>              Set the unicast hop limit for the socket.  Argument is a
>              pointer to an integer.  -1 in the value means use the route
>              default, otherwise it should be between 0 and 255.

**Comment by little-dude** [comment]
Just tried the development version on Archlinux, and the issue is fixed.
However I had to rollback to fa5b51cbc0fee3958abf31c86a570bd625724201. https://github.com/schweikert/fping/commit/2105561146bde0c61d98c64beec16e260b804863 seems to break the build on Arch:
```
~/fping ❯❯❯ ./configure  --enable-ipv4 --enable-ipv6                                             
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for netinet/icmp6.h... yes
configure: error: IPv6 not supported on this platform (netinet/icmp6.h header not found)
```

**Comment by schweikert** [comment]
Thanks for reporting the problem!

Yes, it was fixed in develop (and version3 branch). The fix will be released with version 3.16 soon (probably this week).

I have also fixed the ipv6 issue in develop now.
